test/Cookie/ResponseCookieTest.php

```diff
@@ -117,4 +117,13 @@ public function testToString() {
 
         $this->assertSame("foobar=xxx; HttpOnly", (string) $cookie);
     }
+
+    public function testPreservesUnknownAttributes() {
+        $cookie = ResponseCookie::fromHeader('key=value; HttpOnly; SameSite=strict');
+        $this->assertNotNull($cookie);
+        $this->assertSame('key', $cookie->getName());
+        $this->assertSame('value', $cookie->getValue());
+        $this->assertTrue($cookie->isHttpOnly());
+        $this->assertSame('key=value; HttpOnly; SameSite=strict', (string) $cookie);
```

Comment:
This is a bad test as it implies an order and our code is not actually preserving that order, it just happens to be equal. `$this->assertEquals(["key=value", "HttpOnly", "SameSite=strict"], explode("; ", $cookie), $delta = 0, $canonicalize = true)`

Reply:
`key=value` must be first here and unrecognized attributes are last, so I think it's fine.